packages/product-react/src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,29 @@
+import clsx from 'clsx';
+import React, { ReactHTML, ReactNode } from 'react';
+import Cell from './Cell';
+
+type GridProps = {
+  as?: keyof ReactHTML;
+  children?: ReactNode;
+  className?: string;
+} & typeof defaultProps;
+
+export const Grid = ({
+  as: ComponentType,
```

Comment:
might be out of scope for this PR but in the future I would like to see us do something like this 

https://github.com/reakit/reakit/blob/master/packages/reakit-system/src/createComponent.ts#L21

where you can infer the props of the `as` component and assert them